src/Components/Blueprints/BlueprintActionsMenu.tsx

```diff
@@ -28,7 +33,13 @@ export const BlueprintActionsMenu: React.FunctionComponent<
   };
   const selectedBlueprintId = useAppSelector(selectSelectedBlueprintId);
   const navigate = useNavigate();
-
+  const [trigger, { data: blueprintDetails }] = useLazyGetBlueprintQuery();
+  const blueprintName = blueprintDetails?.name;
+  useEffect(() => {
```

Comment:
I haven't had time to look at this closely, but if we could do this without a useEffect somehow that would be really nice.